src/BSON/Document.c

```diff
@@ -471,6 +491,91 @@ static HashTable* php_phongo_document_get_properties(phongo_compat_object_handle
 	return php_phongo_document_get_properties_hash(object, false, 1);
 }
 
+zval* php_phongo_document_read_property(phongo_compat_object_handler_type* object, phongo_compat_property_accessor_name_type* member, int type, void** cache_slot, zval* rv)
+{
+	php_phongo_document_t* intern;
+	char*                  key;
+	size_t                 key_len;
+
+	intern = Z_OBJ_DOCUMENT(PHONGO_COMPAT_GET_OBJ(object));
+
+	PHONGO_COMPAT_PROPERTY_ACCESSOR_NAME_TO_STRING(member, key, key_len);
+
+	if (!php_phongo_document_get(intern, key, key_len, rv)) {
+		// Exception already thrown
+		return &EG(uninitialized_zval);
```

Comment:
Is this expected by the object handler instead of returning `NULL`? I assume you borrowed this from ArrayObject, so feel free to confirm and resolve on your own.

Reply:
Yes, copied it over. If it seems more consistent I'm also happy to return `NULL`.

Reply:
No, I think `&EG(uninitialized_zval)` is the right call here. I don't know enough about this API to say whether `NULL` would be equivalent.